src/main/java/pl/oskarkowalski/recruitment/ticketbookingapp/enums/TicketPrice.java

```diff
@@ -0,0 +1,31 @@
+package pl.oskarkowalski.recruitment.ticketbookingapp.enums;
+
+import lombok.Getter;
+
+@Getter
+public enum TicketPrice {
+  ADULT((double) 25),
+  STUDENT((double) 18),
+  CHILD(12.5),
+  ADULT_WEEKEND((double) 29),
+  STUDENT_WEEKEND((double) 22),
+  CHILD_WEEKEND(16.5);
+
+  private final Double amount;
+
+  TicketPrice(Double amount) {
+    this.amount = amount;
+  }
+
+  public static TicketPrice fromAmount(Double amount) {
+    if(amount.equals(ADULT.amount)) return ADULT;
+    else if(amount.equals(STUDENT.amount)) return STUDENT;
```

Comment:
trochę nieczytelnie wyszły te ify

Reply:
To prawda - muszę to zrefaktorować - na przykład poprzez użycie domyślnej metody values()